sdk/core/src/pageable.rs

```diff
@@ -82,12 +84,12 @@ impl<T, O> std::fmt::Debug for Pageable<T, O> {
 
 /// A type that can yield an optional continuation token
 pub trait Continuable {
-    fn continuation(&self) -> Option<String>;
+    fn continuation(&self) -> Option<Continuation>;
```

Comment:
Do any endpoints allow for multiple types of continuations? It seems like right now the `Continuation` header with a continuation token is the happy path and we're using panics to ensure that we don't use a `Continuation::Range` when we don't meant time.

What if we were to trying the following;

```rust
pub trait Continuable {
   type Continuation: crate::AsHeaders;
    fn continuation(&self) -> Option<Self::Continuation>;
}
```

The `make_request` function then becomes `impl Fn(Option<T::Continuation>) -> F + Clone + 'static + Send`

This would make the system more robust. Every continuable response just needs to provide the headers it uses for specifying how it continues.

Reply:
Examples of different Continuation methods:

* BlobClient.get uses Range header
* Cosmos's list_attachments uses `Continuation` header
* ContainerClient.list_blobs uses `marker` URI parameter
* datalake's file_systems_list uses `maker` URI parameter.  Note, this one also accepts next_marker from the builder, which probably shouldn't be available.